docs/tutorials/start-dev-with-ralph.md

```diff
@@ -0,0 +1,348 @@
+# How to start developing with Ralph ?
+
+### Prepare your working environment 
+
+To start developing with Ralph, you must install the following dependencies on your local environment: 
+
+- **Python 3.11**
+- make
+- Docker/ Docker compose
+
+### Bootstrap Ralph for development
+
+To start playing with `ralph`, you should build it using the `bootstrap` Make target:
+
+```bash
+make bootstrap
+```
+
+Once the project has been bootstrapped, you may want to edit generated `.env`
+file to set up available backends parameters that will be injected in the
+running container as environment variables to configure Ralph (see [backends
+documentation](./backends.md)):
+
+```bash
+# Elasticsearch backend
+RALPH_ES_HOSTS=http://elasticsearch:9200
+RALPH_ES_INDEX=statements
+RALPH_ES_TEST_HOSTS=http://elasticsearch:9200
+RALPH_ES_TEST_INDEX=test-index
+
+# [...]
+```
+
+> Note that lines starting with a `#` are considered as commented and thus will
+> have no effect while running Ralph.
+
+### Working with backends
+
+To configure the backends, we provide default parameters in the `.env.dist`
+template, you can copy/paste them in your `.env` file (and uncomment them so
+that they are properly injected in running containers).
+
+> In order to run the Elasticsearch backend locally on GNU/Linux operating
+> systems, ensure that your virtual memory limits are not too low and increase
+> them (temporally) if needed by typing this command from your terminal (as
+> `root` or using `sudo`):
+>
+> `sysctl -w vm.max_map_count=262144`
+>
+> Reference:
+> https://www.elastic.co/guide/en/elasticsearch/reference/master/vm-max-map-count.html
+
```

Comment:
Perhaps we should also mention that we need at least 10% of remaining disk space to run Elasticsearch ?